packages/devtools_server/lib/src/handlers.dart

```diff
@@ -36,10 +36,50 @@ Future<shelf.Handler> defaultHandler() async {
 
   // Make a handler that delegates based on path.
   return (shelf.Request request) {
+    // The API handler takes all calls to api/.
+    if (Api.canHandle(request)) {
+      return Api.handle(request);
+    }
     return request.url.path.startsWith('packages/')
         // request.change here will strip the `packages` prefix from the path
         // so it's relative to packHandler's root.
         ? packHandler(request.change(path: 'packages'))
         : buildHandler(request);
   };
 }
+
+/// The DevTools server API.
+///
+/// This defines endpoints that serve all requests that come in over api/.
+class Api {
+  /// Determines whether or not [request] is an API call.
+  static bool canHandle(shelf.Request request) {
+    return request.url.path.startsWith('api/');
+  }
+
+  /// Handles all requests.
+  ///
+  /// To override an API call, pass in a subclass of [Api].
+  static FutureOr<shelf.Response> handle(shelf.Request request, [Api api]) {
+    api ??= Api();
+    switch (request.url.path) {
+      case 'api/logPageView':
+        return api.logPageView(request);
+      default:
+        return api.notImplemented;
+    }
+  }
+
+  /// Logs a page view in the DevTools server.
+  ///
+  /// In the open-source version of DevTools, Google Analytics handles this
+  /// without any need to involve the server.
+  FutureOr<shelf.Response> logPageView(shelf.Request request) async {
+    return notImplemented;
```

Comment:
where is this called? Is there a reason not to have a no-op implementation instead of having this throw?

Reply:
This is called from `packages/devtools_app/lib/src/ui/analytics.dart` every time there is a `ga.screen()` invocation (that is, a screen change), the web app forwards the invocation to the server.

Reply:
ok sounds good.